File: WindowsFormsApplication3b/MainForm.cs
using System;
using System.Collections.Generic;
using System.Net;
using System.Text.RegularExpressions;
using System.Windows.Forms;

namespace WindowsFormsApplication3b
{
	public partial class MainForm : Form
	{
		public MainForm()
		{
			InitializeComponent();
		}

		private WebClient webClient;

		private void getHtmlButton_Click(object sender, EventArgs e)
		{
			listBox.Items.Clear();
			webClient = new WebClient();
			webClient.DownloadProgressChanged += (o, args) =>
			                                  {
			                                  	progressBar.Value = args.ProgressPercentage;
			                                  };
			webClient.
				DownloadStringCompleted +=
				(o, args) =>
				{
					getHtmlButton.Enabled = true;
					cancelButton.Visible = false;
					webClient.Dispose();
					webClient = null;
					if (!args.Cancelled)
					{
						foreach (string x in GetScriptBodies(args.Result))
							listBox.Items.Add(x);
					}
					else
					{
						MessageBox.Show("Cancelled");
						progressBar.Visible = false;
					}
				};
			progressBar.Visible = true;
			cancelButton.Visible = true;
			progressBar.Value = 0;
			webClient.DownloadStringAsync(new Uri("http://google.ca"));

			getHtmlButton.Enabled = false;
		}

		private void cancelButton_Click(object sender, EventArgs e)
		{
			if(webClient != null) webClient.CancelAsync();
		}

		private static IEnumerable<string> GetScriptBodies(string html)
		{
			const string pattern = @"<script.*?>\s*(?'scriptBody'.+?)\s*</script>";
			MatchCollection matches = Regex.Matches(html, pattern);
			if (matches.Count == 0) yield break;
			foreach (Match match in matches)
			{
				string scriptBody = match.Groups["scriptBody"].Value;
				yield return scriptBody;
			}
			yield break;
		}
	}
}

File: WindowsApplication3b/MainForm.vb
Imports System.Text.RegularExpressions
Imports System.Net

Public Class MainForm
    Private webClient As WebClient

    Private Sub getHtmlButton_Click(sender As System.Object, e As System.EventArgs) Handles getHtmlButton.Click
        listBox.Items.Clear()
        webClient = New WebClient()
        AddHandler webClient.DownloadProgressChanged,
            Sub(o As Object, args As DownloadProgressChangedEventArgs) progressBar.Value = args.ProgressPercentage

        AddHandler webClient.DownloadStringCompleted,
            Sub(o As Object, args As DownloadStringCompletedEventArgs)
                getHtmlButton.Enabled = True
                cancelButton.Visible = False
                webClient.Dispose()
                webClient = Nothing
                If (args.Cancelled) Then
                    MessageBox.Show("Cancelled")
                    progressBar.Visible = False
                Else
                    For Each x As String In GetScriptBodies(args.Result)
                        listBox.Items.Add(x)
                    Next
                End If
            End Sub

        progressBar.Visible = True
        cancelButton.Visible = True
        progressBar.Value = 0
        webClient.DownloadStringAsync(New Uri("http://google.ca"))
        getHtmlButton.Enabled = False
    End Sub

    Private Sub cancelButton_Click(sender As Object, e As EventArgs) Handles cancelButton.Click
        If (Not webClient Is Nothing) Then
            webClient.CancelAsync()
        End If
    End Sub

    Private Shared Function GetScriptBodies(ByVal html As String) As IEnumerable
        Const pattern As String = "<script.*?>\s*(?'scriptBody'.+?)\s*</script>"
        Dim matches As MatchCollection = Regex.Matches(html, pattern)
        Dim result(matches.Count - 1) As String
        Dim i As Integer = 0
        For Each match As Match In matches
            Dim scriptBody As String = match.Groups("scriptBody").Value
            Dim num As Integer = i
            i = num + 1
            result(num) = scriptBody
        Next
        Return result
    End Function

End Class
